src/Sparrow/Json/ArenaMemoryAllocator.cs

```diff
@@ -54,13 +52,19 @@ public long Allocated
             }
         }
 
-        public ArenaMemoryAllocator(int initialSize = 1024 * 1024)
+        public ArenaMemoryAllocator(LowMemoryFlag lowMemoryFlag, int initialSize = 1024 * 1024)
         {
+            if (lowMemoryFlag.LowMemoryState != 0 && initialSize > 1024 * 1024)
+            {                
+                Console.WriteLine("Avoiding arena creation");
+                throw new LowMemoryException("Will not renew arena due to low memory stress");
```

Comment:
You aren't handling throwing of this exception properly.
When the exception is thrown, we need to return `503`, not `500`